addons/misc/functions/fnc_suitcaseNukeZeus.sqf

```diff
@@ -0,0 +1,211 @@
+#include "script_component.hpp"
+/*/////////////////////////////////////////////////
+Author: Landric
+               
+File: fnc_suitcaseNukeZeus.sqf
+Parameters: pos
+Return: none
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3]];
+
+private _onConfirm =
+{
+    params ["_dialogResult","_in"];
+    _in params [["_pos",[0,0,0],[[]],3]];
+    _dialogResult params [
+        "_countdownLength",
+        "_effect",
+        "_defuser",
+        "_defuseTime",
+        ["_code", "", [""]],
+        ["_target", 0, [0]]
+    ];
+
+    private _device = "Land_MultiScreenComputer_01_black_F" createVehicle _pos;
+    ["zen_common_updateEditableObjects", [[_device]]] call CBA_fnc_serverEvent;
+    _device setVariable [QGVAR(suitcaseNuke_isArmed), true, true];
+
+
+    // Timer code
+    // TODO: move to server to avoid problems if zeus crashes?
+    [_device, _countdownLength] spawn { 
+
+        params ["_device","_countdownLength"]; 
+
+        private "_activateTime";
+        private "_countdownRemaining";
+        // Not strictly needed, but makes it possible to test in SP!
+        if(isMultiplayer) then {
+            _activateTime = serverTime + _countdownLength;
+            _countdownRemaining = _activateTime - serverTime;
+        } else {
+            _activateTime = time + _countdownLength;
+            _countdownRemaining = _activateTime - time;
+        };
+
+        private "_countdownRemainingText";
+        private _isArmed = true;
+        while {alive _device and {_isArmed and {_countdownRemaining > 0}}} do { 
+            
+            _countdownRemainingText = [_countdownRemaining, "MM:SS"] call BIS_fnc_secondsToString; 
+            { 
+                _device setObjectTextureGlobal[_x, 
+                    format ['#(rgb,512,512,3)text(0,1,"LCD14",0.3,"#171717","#eb4034","\n%1")', _countdownRemainingText]
+                ]; 
+            } forEach [1,2,3];
+
+            if(isMultiplayer) then {
+                _countdownRemaining = _activateTime - serverTime;
+            } else {
+                _countdownRemaining = _activateTime - time;
+            };
+            if(_countdownRemaining < 0) then { _countdownRemaining = 0; };
+
+            _isArmed = _device getVariable [QGVAR(suitcaseNuke_isArmed), false];
+
+            sleep 0.3; 
+        };
+
+        if(_isArmed) then {
+            _device setDamage 1;
+        };
+
+        while {not isNull _device} do {
+            // if(_countdownRemaining > 0) then {
+                { 
+                    _device setObjectTextureGlobal[_x, 
+                        format ['#(rgb,512,512,3)text(0,1,"LCD14",0.3,"#171717","#eb4034","\n%1")', _countdownRemainingText]
+                    ]; 
+                } forEach [1,2,3];
```

Comment:
indexing not correct if the rest is meant to be removed/commented out.

Reply:
How d'you mean?

Reply:
yeah, was meaning indentation. Fair. I haven't looked into code commented out, I just skipped those parts 😅 

A custom message could be nice as optional for when hitting zero, but unsure if anyone would actually see it as the effect would fire at 0. So I will leave it up to you what you think makes sense.

Reply:
Well if it deploys smoke or EMP, the message would remain on-screen. But there'd be enough bother getting the font-size correct that its probably not worth it for arbitrary messages when most people won't see it